distsql/select_result.go

```diff
@@ -145,6 +178,27 @@ func (r *selectResult) NextRaw(ctx context.Context) (data []byte, err error) {
 // Next reads data to the chunk.
 func (r *selectResult) Next(ctx context.Context, chk *chunk.Chunk) error {
 	chk.Reset()
+	// Check the returned data is default/arrow format.
+	if r.selectResp == nil || len(r.selectResp.RowBatchData) == 0 {
+		err := r.getSelectResp()
+		if err != nil || r.selectResp == nil {
+			return errors.Trace(err)
+		}
+		if len(r.selectResp.RowBatchData) == 0 {
+			r.decodeType = DecodeTypeDefault
```

Comment:
should we add a metrics here when the expected decode type is `arrow` but the actual decode type is `default`?

Reply:
Ok.